components/brave_wallet/browser/siwe_message_parser.cc

```diff
@@ -0,0 +1,402 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_wallet/browser/siwe_message_parser.h"
+
+#include <vector>
+
+#include "base/strings/strcat.h"
+#include "base/strings/string_tokenizer.h"
+#include "base/strings/string_util.h"
+#include "brave/components/brave_wallet/common/eth_address.h"
+#include "brave/components/brave_wallet/common/sign_message_request.mojom.h"
+#include "url/gurl.h"
+#include "url/origin.h"
+#include "url/url_constants.h"
+
+using base::StringTokenizer;
+
+namespace brave_wallet {
+
+namespace {
+constexpr size_t kMinNonceLength = 8;
+constexpr char kStartingToken[] =
+    " wants you to sign in with your Ethereum account:";
+constexpr char kURIToken[] = "URI: ";
+constexpr char kVersionToken[] = "Version: ";
+constexpr char kChainIdToken[] = "Chain ID: ";
+constexpr char kNonceToken[] = "Nonce: ";
+constexpr char kIssuedAtToken[] = "Issued At: ";
+constexpr char kExpirationTimeToken[] = "Expiration Time: ";
+constexpr char kNotBeforeToken[] = "Not Before: ";
+constexpr char kRequestIdToken[] = "Request ID: ";
+constexpr char kResourcesToken[] = "Resources:";
+constexpr char kResourcesSeperator[] = "- ";
+
+bool ConsumeDelim(StringTokenizer* tokenizer) {
+  CHECK(tokenizer);
```

Comment:
Just double checking my understanding here. If this check fails will line 40 crash the browser?

Reply:
Yes, and that is intentional which means this function won't be called with nullptr.